showq/showq.go

```diff
@@ -133,6 +145,18 @@ func (s *Showq) collectBinaryShowqFromScanner(file io.Reader) error {
 	scanner.Split(ScanNullTerminatedEntries)
 	queueSizes := make(map[string]float64)
 
+	// Remove metrics from size and age histograms and re-initialize them. HistogramVec
+	// is intended to capture data streams. Showq however always returns all emails
+	// currently queued, therefore we need to reset the histograms before every collect.
+	for _, q := range []string{"active", "deferred", "hold", "incoming", "maildrop"} {
```

Comment:
Thank you for your detailed report and pull request.

`Showq` keeps track of encountered queues in the `knownQueues` field, so please use that instead of hardcoded values.

Additionally, `DeleteLabelValues` is dependend on the order of labels, so while this currently works, it does not sound very maintainable. Since metrics are re-created as they are collected, consider using `Reset` instead.